### PR TITLE
feat(backend): clear latents files on startup

### PR DESCRIPTION
**Pull request by psychedelicious**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Description

[feat(backend): clear latents files on startup](https://github.com/invoke-ai/InvokeAI/pull/4704/commits/c3b842381a71a66b226c120ef8f7b8f3c50c0d40)
[c3b8423](https://github.com/invoke-ai/InvokeAI/pull/4704/commits/c3b842381a71a66b226c120ef8f7b8f3c50c0d40)

Adds logic to `DiskLatentsStorage.start()` to empty the latents folder on startup.

Adds start and stop methods to `ForwardCacheLatentsStorage`. This is required for `DiskLatentsStorage.start()` to be called, due to how this particular service breaks the direct DI pattern, wrapping the underlying storage with a cache.

## Related Tickets & Documents

I'm sure there are some but I can't find them

### CODE REVIEW

File: invokeai/app/api/dependencies.py [hunk 1]
```diff
@@ -122,6 +124,14 @@ def initialize(config: InvokeAIAppConfig, event_handler_id: int, logger: Logger
 
         db.clean()
 
+        # clean up latents - this is a temporary measure to respect users' disk space
+        deleted_latents_count = 0
+        for latents in Path(latents_path).glob("*"):
+            if latents.is_file():
+                latents.unlink()
```
Comment (maryhipp) [inline invokeai/app/api/dependencies.py:131]: Does unlink actually delete also?

Comment (psychedelicious) [reply invokeai/app/api/dependencies.py:131]: Yes

Comment (Neosettler) [reply invokeai/app/api/dependencies.py:131]: Please, consider moving these files outside the "output" folders. Also, cleanliness is next to godliness, don't just remove the files, delete parent directory instead. No one wants empty folders either.

follow-up on Discord channel:
https://discord.com/channels/1020123559063990373/1156344991292063825/1156344991292063825

Comment (psychedelicious) [reply invokeai/app/api/dependencies.py:131]: - They *are* outputs, so they will be in the outputs folder, which is primarily an application-facing folder.
- The `latents/` directory is used by the application, so it needs to exist.
- We are using modern python's `pathlib` as a general rule. `pathlib` does not delete non-empty directories. Globbing the files and deleting each one is a fine way to handle this.

**Comment by psychedelicious** [comment]
Marking as draft as this is waiting on #4671

**Comment by psychedelicious** [comment]
Forgot about this - fixed it up.

Output looks like this:
```sh
[2023-11-30 20:07:34,705]::[InvokeAI]::INFO --> Deleted 256 latents files (freed 8.61MB)
[2023-11-30 20:07:34,714]::[InvokeAI]::INFO --> Pruned 127 finished queue items
[2023-11-30 20:07:34,789]::[InvokeAI]::INFO --> Cleaned database (freed 0.56MB)
```

**Pull request merged**